src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRanger.java

```diff
@@ -102,10 +98,32 @@ public AIState getAttackState()
         return GUARD_ATTACK_RANGED;
     }
 
+    /**
+     * Blockdistance at which attackRanged() will get Control, intentionally set higher than actual attack range
+     */
     @Override
     protected int getAttackRange()
     {
-        return (MAX_DISTANCE_FOR_ATTACK / ((buildingGuards.getBuildingLevel() / buildingGuards.getMaxBuildingLevel()) + 1)) + 10;
+        return getRealAttackRange() + 10;
+    }
+
+    /**
+     * Calculates the actual attack range
+     */
+    private int getRealAttackRange()
+    {
+        int attackDist = BASE_DISTANCE_FOR_RANGED_ATTACK;
+        // + 2 Blockrange per building level for a total of +10 from building level
+        if (buildingGuards != null)
+        {
+            attackDist += buildingGuards.getBuildingLevel() * 2;
+        }
+        // ~ +1 each three levels for a total of +15 from guard level
+        if (worker.getCitizenData() != null)
```

Comment:
I like how it scales.

Reply:
:) figured sclaing with hut level + guard level would be nice